include/alpaka/core/RuntimeMacros.hpp

```diff
@@ -0,0 +1,45 @@
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+#pragma once
+
+//! ALPAKA_DEVICE_THROW either aborts(terminating the program and creating a core dump) or throws std::runtime_error
+//! depending on the Acc. The std::runtime_error exception can be catched in the catch block.
+//!
+//! CUDA __trap function triggers std::runtime_error but can be catched during wait not exec.
+//!
+//! The OpenMP specification mandates that exceptions thrown by some thread must be handled by the same thread.
+//! Therefore std::runtime_error thrown by ALPAKA_DEVICE_THROW aborts the the program for OpenMP backends. If needed
+//! the SIGABRT signal can be catched by signal handler.
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED) && defined(__CUDA_ARCH__)
+#    define ALPAKA_DEVICE_THROW(MSG)                                                                                  \
+        {                                                                                                             \
+            printf(                                                                                                   \
+                "alpaka encountered a user-defined error condition while running on the Cuda back-end: \n %s ",       \
+                (MSG));                                                                                               \
+            __trap();                                                                                                 \
+        }
+#elif defined(ALPAKA_ACC_GPU_HIP_ENABLED) && defined(__HIP_DEVICE_COMPILE__)
+#    define ALPAKA_DEVICE_THROW(MSG)                                                                                  \
+        {                                                                                                             \
+            printf(                                                                                                   \
+                "alpaka encountered a user-defined error condition while running on the HIP back-end: \n %s ",        \
+                (MSG));                                                                                               \
+            abort();                                                                                                  \
+        }
+#elif defined(ALPAKA_ACC_SYCL_ENABLED) && defined(__SYCL_DEVICE_ONLY__)
+#    define ALPAKA_DEVICE_THROW(MSG)                                                                                  \
+        {                                                                                                             \
+            printf(                                                                                                   \
+                "alpaka encountered a user-defined error condition while running on the Sycl back-end: \n %s ",       \
+                (MSG));                                                                                               \
```

Comment:
I don't think `printf` works like this in SYCL code running on Intel GPUs.

Reply:
you need to include `alpaka/core/Sycl.hpp` for the `printf` to work in SYCL

Reply:
Thanks, I removed the line.

Reply:
I think you can keep this line and add the include, it's useful to have a message that explains the reason for the throw.

Reply:
Ok, done. Thanks.